### ISSUE TITLE
Rule: Invalid Regular Expression literal

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#346]
We need a rule to determine if a regular expression created via constructor is valid or not. The JS engine will pick up syntax errors in regular expression literals, but not constructors. Example:

``` js
var re = new RegExp("[");
```

The rule ID should be `valid-regex` and be on by default.
